src/core/algorithms/gfd/gfd_handler.h

```diff
@@ -0,0 +1,42 @@
+#pragma once
+#include <vector>
+
+#include "algorithms/algorithm.h"
+#include "config/names_and_descriptions.h"
+#include "gfd.h"
+#include "parser.h"
+
+namespace algos {
+
+class GfdHandler : public Algorithm {
+protected:
+    std::filesystem::path graph_path_;
+    std::vector<std::string> gfd_paths_;

```

Comment:
Why was it decided to directly receive paths from options? In the case when we work with CSV tables, the logic for creating tables is moved outside the algorithm into special options. (`config::InputTable`)

Reply:
I decided not to clutter the code with new objects, since these fields are very simple. I think it's pretty natural to supply the paths directly

Reply:
Thanks for the answer. The motivation is clear, but it seems to me that code consistency is more important. But I would like to hear an opinion from @polyntsov.

Reply:
The reason we have a separator option `Table` for CSV tables is because the table can be specified for the algorithm not only as path, but also as pandas dataframe. So what we need to think about here is how do we want to specify input graphs from python, if using path to the file, then storing only path is ok. 
Since in this PR we're not touching python bindings at all, there is little reason to create a separate option, and we can just store paths for simplicity. So IMO the separate option, if required, should be added in the future along with python bindings.